src/test/java/com/sap/sgs/phosphor/fosstars/model/score/AverageCompositeScoreTest.java

```diff
@@ -0,0 +1,176 @@
+package com.sap.sgs.phosphor.fosstars.model.score;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+import com.sap.sgs.phosphor.fosstars.model.Confidence;
+import com.sap.sgs.phosphor.fosstars.model.Feature;
+import com.sap.sgs.phosphor.fosstars.model.Score;
+import com.sap.sgs.phosphor.fosstars.model.Value;
+import com.sap.sgs.phosphor.fosstars.model.feature.DoubleFeature;
+import com.sap.sgs.phosphor.fosstars.model.value.ScoreValue;
+import org.junit.Test;
+
+public class AverageCompositeScoreTest {
+
+  private static final double PRECISION = 0.001;
+
+  @Test
+  public void smokeTest() {
+    TestAverageCompositeScore score = new TestAverageCompositeScore();
+    assertEquals(TestAverageCompositeScore.NAME, score.name());
+    assertEquals(2, score.allFeatures().size());
+    assertTrue(score.allFeatures().contains(FirstScore.FEATURE));
+    assertTrue(score.allFeatures().contains(SecondScore.FEATURE));
+    assertTrue(score.features().isEmpty());
```

Comment:
Please correct me if I am wrong. What I understand here is 
- score.allFeatures() indicate all features including sub-score features().
- score.features() indicates only the features of the root score, which is actually composed by sub-scores.

Reply:
`Score.allFeatures()` returns all features which are used by a score. But It doesn't return sub-scores. `Score.features()` returns features which are directly used by a score.

Reply:
Got it